lambdajs/src/SharedAbstract.hs

```diff
@@ -0,0 +1,262 @@
+{-# LANGUAGE Arrows                     #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE FunctionalDependencies     #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE TypeSynonymInstances       #-}
+module SharedAbstract where
+
+import           Derivations
+import           GHC.Generics                                    (Generic)
+import           Prelude                                         hiding (break,
+                                                                  error, fail,
+                                                                  lookup, map,
+                                                                  read)
+import qualified Prelude
+import           SharedInterpreter
+import           Syntax
+
+import           Data.Hashable
+import           Data.Identifiable
+import           Data.Set
```

Comment:
Use `Data.Abstract.DiscretePowerset` that implements all the ordering and least upper bound operations.